models/incremental/fct_dbt_source_freshness.sql

```diff
@@ -0,0 +1,29 @@
+{{ config( materialized='incremental', unique_key='source_freshness_id' ) }}
+
+with sources as (
+
+    select *
+    from {{ ref('int_dbt_sources') }}
+
+),
+
+source_freshness_executions as (
+
+    select *
+    from {{ ref('int_dbt_source_freshness') }}
+
+),
+
+source_freshness_executions_incremental as (
+
+    select *
+    from source_freshness_executions
+
+    {% if is_incremental() %}
+    -- this filter will only be applied on an incremental run
+    where artifact_generated_at > (select max(artifact_generated_at) from {{ this }})
+    {% endif %}
+
+)
+
+select * from source_freshness_executions_incremental
```

Comment:
you removed the join between `sources` and `source_freshness` - what is `int_dbt_sources` doing now